rust/src/transformations/quantile_score_candidates/make_quantile_score_candidates.tex

```diff
@@ -0,0 +1,318 @@
+        \\&\hspace{1cm}\mathrm{abs\_diff}(\alpha_{den} \cdot \min(\#(X' < C_i), l), \alpha_{num} \cdot \min(|X'| - \#(X' = C_i), l))| \\
+    &= \alpha_{den} \cdot \max\limits_{i} ||\min(\#(X < C_i), l) - \alpha \cdot \min(|X| - \#(X = C_i), l)|
+        \\&\hspace{1cm}|\min(\#(X' < C_i), l) - \alpha \cdot \min(|X'| - \#(X' = C_i), l)|| \\
+    &= \alpha_{den} \cdot \max\limits_{i} ||\#(X < C_i) - \alpha \cdot (|X| - \#(X = C_i))|
+        \\&\hspace{1cm}|\#(X' < C_i) - \alpha \cdot (|X| - \#(X' = C_i))|| \\
+\intertext{Consider each of the four cases of changing a row in X.}
+\intertext{Case 1. Assume $X'$ is equal to $X$, but with some $X_j < C_i$ replaced with $X'_j > C_i$.}
+    &= 2 \cdot \alpha_{den} \cdot \max\limits_{i} ||(1 - \alpha) \cdot \#(X < C_i) - \alpha \cdot \#(X > C_i)| 
+        \\&\hspace{1cm} - (1 - \alpha) \cdot (\#(X < C_i) - 1) - \alpha \cdot (\#(X > C_i) + 1)|| &&\text{by definition of }\function \\
+    &\leq 2 \cdot \alpha_{den} \cdot \max\limits_{i} ||(1 - \alpha) \cdot \#(X < C_i) - \alpha \cdot \#(X > C_i)|
+        \\&\hspace{1cm} - (|(1 - \alpha) \cdot \#(X < C_i) - \alpha \cdot \#(X > C_i)| + |1|)| &&\text{by triangle inequality} \\
+    &= 2 \cdot \alpha_{den} \cdot \max\limits_{i} |1| &&\text{scores cancel}\\
+    &= 2 \cdot \alpha_{den}
+\intertext{Case 2. Assume $X'$ is equal to $X$, but with some $X_j > C_i$ replaced with $X'_j < C_i$.}
+    &= 2 \cdot \alpha_{den} &&\text{by symmetry, follows from Case 1.}
+\intertext{Case 3. Assume $X'$ is equal to $X$, but with some $X_j \neq C_i$ replaced with $C_i$.}
+    &\leq 2 \cdot \max(\alpha_{num}, \alpha_{den} - \alpha_{num}) &&\text{equivalent to one removal (see \rustdoc{transformations/fn}{make\_quantile\_score\_candidates})} \\
+\intertext{Case 4. Assume $X'$ is equal to $X$, but with some $X_j = C_i$ replaced with $X'_j \neq C_i$.}
+    &\leq 2 \cdot \max(\alpha_{num}, \alpha_{den} - \alpha_{num}) &&\text{equivalent to one addition (see \rustdoc{transformations/fn}{make\_quantile\_score\_candidates})} \\
+\intertext{Take the union bound over all cases.}
+    d_\infty(s_i, s'_i) &\le \max(2 \cdot \alpha_{den}, 2 \cdot \max(\alpha_{num}, \alpha_{den} - \alpha_{num})) = 2 \cdot \alpha_{den} &&\text{since } \max(\alpha, 1 - \alpha) \leq 1
+\end{align*}
+\end{proof}
+
+Take any two elements $X, X'$ in the \texttt{input\_domain} and any pair $(\din, \dout)$, 
+where \din\ has the associated type for \texttt{input\_metric} and \dout\ has the associated type for \texttt{output\_metric}.
+Assume $X, X'$ are \din-close under \texttt{input\_metric} and that $\texttt{stability\_map}(\din) \leq \dout$. 
+
+\begin{align*}
+    \dout &= \max\limits_{X \sim X'} d_{IDD}(s, s') \\
+    &= \max\limits_{X \sim X'} \max\limits_{ij} |(s_i - s'_i) - (s_j - s'_j)| &&\text{by definition of }\rustdoc{metrics/struct}{LInfDiffDistance} \\
+    &\leq 2 \max\limits_{X \sim X'} \max\limits_{i} |s_i - s'_i| &&\text{$s$ is not monotonic; take looser bound}\\
```

Comment:
and here also (maybe I just don't have the background for it). is there a link ?